lms/djangoapps/instructor/views/api.py

```diff
@@ -2629,8 +2636,20 @@ def spoc_gradebook(request, course_id):
         for student in enrolled_students
     ]
```

Comment:
This was exactly my concern. The point of pagination is that we should only be loading `MAX_STUDENTS_PER_PAGE_GRADE_BOOK` students at a time. If we're still loading all of them, pagination is not doing its job.